cpp/src/arrow/ipc/json-simple.cc

```diff
@@ -142,10 +141,60 @@ class BooleanConverter final : public ConcreteConverter<BooleanConverter> {
 
   std::shared_ptr<ArrayBuilder> builder() override { return builder_; }
 
- protected:
+ private:
   std::shared_ptr<BooleanBuilder> builder_;
 };
 
+// ------------------------------------------------------------------------
+// Helpers for numeric converters
+
+// Convert single signed integer value (also {Date,Time}{32,64} and Timestamp)
+template <typename T, typename CType = typename TypeTraits<T>::CType>
+enable_if_signed_integer<typename CTypeTraits<CType>::ArrowType, Status> ConvertNumber(
```

Comment:
This function handles date, time, and timestamp types as well (which are represented with integers but do not inherit `Integer`). If it would be more clear, I could write something like
```suggestion
enable_if<is_signed_integer<ArrowType>::value
       || is_date<ArrowType>::value
       || is_time<ArrowType>::value
       || is_timestamp<ArrowType>::value,
    Status> ConvertNumber(
```

Reply:
Indeed, that would make the intent more explicit.